cmake_file_api/cmake.py

```diff
@@ -4,20 +4,32 @@
 
 from .reply.api import REPLY_API
 
+PathLike = Union[Path, str]
+
 
 class CMakeProject(object):
     __slots__ = ("_source_path", "_build_path", "_api_version", "_cmake")
 
-    def __init__(self, build_path: Union[Path, str], source_path: Optional[Union[Path, str]]=None, api_version: Optional[int]=None, cmake: Optional[str]=None):
+    def __init__(self, build_path: PathLike, source_path: Optional[PathLike]=None, api_version: Optional[int]=None, cmake: Optional[str]=None):
         if not build_path:
             raise ValueError("Need a build folder")
-        if isinstance(source_path, str):
-            source_path = Path(source_path).resolve()
-        self._source_path = source_path.resolve() if source_path else None
         if isinstance(build_path, str):
             build_path = Path(build_path).resolve()
         self._build_path = build_path
+
+        cache = self.build_path / "CMakeCache.txt"
+
+        if source_path is None and cache.exists():
+            source_path = self._cache_lookup("CMAKE_HOME_DIRECTORY", cache)
```

Comment:
Great idea about using the cache to initialize.
But the docs say to prefer `CMAKE_SOURCE_DIR` over `CMAKE_HOME_DIRECTORY`.

Reply:
`CMAKE_SOURCE_DIR` is not a cache variable. I assume it is aliased at CMake runtime from `CMAKE_HOME_DIRECTORY`. Per the docs,

> [`CMAKE_HOME_DIRECTORY`] is an internal cache entry used to locate the source directory when loading a CMakeCache.txt from a build tree.

https://cmake.org/cmake/help/latest/variable/CMAKE_HOME_DIRECTORY.html

Reply:
Thanks, it makes sense now!